packages/shared/src/logger.ts

```diff
@@ -4,9 +4,7 @@ import LokiTransport from "winston-loki";
 import { AuthInfo } from "./graphql/fetchAuthInfoFromGraphQL";
 import { getDeviceId } from "./getDeviceId";
 import { randomUUID } from "crypto";
-
-// Does not work with import. Only works with require().
-const StackUtils = require("stack-utils");
+import StackUtils from "stack-utils";
```

Comment:
Hmm. Did you test this? Looking at the `stack-utils` package ... this won't work. (There is no default export. They set `module.exports = StackUtils` (which is the class)

Reply:
I did... but this is a great point. I'd expect something to be veeeery wrong already. I think we lucked out because we don't depend on many packages that we import defaults from. Any package that would contain `module.exports.default` would fail now with the current setup.

We can see here what is the output of this code right now ([rollup repl](https://rollupjs.org/repl/?version=4.18.0&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMFN0YWNrVXRpbHMlMjBmcm9tJTIwJ3N0YWNrLXV0aWxzJyU1Q25jb25zb2xlLmxvZyhTdGFja1V0aWxzKSUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIyaW50ZXJvcCUyMiUzQSUyMmRlZmF1bHQlMjIlN0QlMkMlMjJ0cmVlc2hha2UlMjIlM0F0cnVlJTdEJTdE)):
```ts
'use strict';

var StackUtils = require('stack-utils');

console.log(StackUtils);
```

This is clearly wrong for the mentioned cases with `.default`. We should be using `interop: 'auto'` to fix this. With that option the output would look like this ([rollup repl](https://rollupjs.org/repl/?version=4.18.0&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMFN0YWNrVXRpbHMlMjBmcm9tJTIwJ3N0YWNrLXV0aWxzJyU1Q25jb25zb2xlLmxvZyhTdGFja1V0aWxzKSUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIyaW50ZXJvcCUyMiUzQSUyMmF1dG8lMjIlN0QlMkMlMjJ0cmVlc2hha2UlMjIlM0F0cnVlJTdEJTdE)):
```ts
'use strict';

var StackUtils = require('stack-utils');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var StackUtils__default = /*#__PURE__*/_interopDefault(StackUtils);

console.log(StackUtils__default.default);
```

Reply:
I had also tried this at one point (seeing if an `import` would trigger our build validation error, where `require` did not). It didn't, but `StackUtils` threw an error when instantiated with `new` because the thing that was imported wasn't a class. I'm...surprised this didn't throw for you.

Reply:
Now I got scared that I didn't test it properly. I'll double-check before merging.

The reason why it just couldn't throw our validation error was the require usage here - those are not special in vanilla Rollup. It's an ESM bundler and CJS support is only brought via plugins. So it didn't even see that it was being loaded by anything as only actual imports are resolved by default.

But we also had those other issues in this plugin here that prevented this from doing its job even after switching to `import`.

Reply:
I re-confirmed that it works now.

Reply:
Okay cool. I wonder why it didn't work when I tried it.